File: features/merchant/reputation/src/main/java/com/tokopedia/tkpd/tkpdreputation/review/shop/view/adapter/ReviewShopTypeFactoryAdapter.java
package com.tokopedia.tkpd.tkpdreputation.review.shop.view.adapter;

import android.view.View;

import com.tokopedia.abstraction.base.view.adapter.viewholders.AbstractViewHolder;
import com.tokopedia.tkpd.tkpdreputation.review.product.view.adapter.ReviewProductContentViewHolder;
import com.tokopedia.tkpd.tkpdreputation.review.product.view.adapter.ReviewProductTypeFactoryAdapter;

/**
 * Created by zulfikarrahman on 1/19/18.
 */

public class ReviewShopTypeFactoryAdapter extends ReviewProductTypeFactoryAdapter {

    private ReviewShopViewHolder.ShopReviewHolderListener shopReviewHolderListener;

    public ReviewShopTypeFactoryAdapter(ReviewProductContentViewHolder.ListenerReviewHolder viewListener,
                                        ReviewShopViewHolder.ShopReviewHolderListener shopReviewHolderListener) {
        super(viewListener);
        this.shopReviewHolderListener = shopReviewHolderListener;
    }

    @Override
    public AbstractViewHolder createViewHolder(View parent, int type) {
        if(type == ReviewShopViewHolder.LAYOUT){
            return new ReviewShopViewHolder(parent, viewListener, shopReviewHolderListener);
        }else{
            return super.createViewHolder(parent, type);
        }
    }

    public int type(ReviewShopModelContent shopReviewModelContent) {
        return ReviewShopViewHolder.LAYOUT;
    }
}

File: features/merchant/shop_page/src/main/java/com/tokopedia/shop/review/shop/view/adapter/ReviewShopTypeFactoryAdapter.kt
package com.tokopedia.shop.review.shop.view.adapter

import android.view.View
import com.tokopedia.abstraction.base.view.adapter.Visitable
import com.tokopedia.abstraction.base.view.adapter.viewholders.AbstractViewHolder
import com.tokopedia.shop.review.product.view.adapter.ReviewProductContentViewHolder
import com.tokopedia.shop.review.product.view.adapter.ReviewProductTypeFactoryAdapter

/**
 * Created by zulfikarrahman on 1/19/18.
 */
class ReviewShopTypeFactoryAdapter(viewListener: ReviewProductContentViewHolder.ListenerReviewHolder,
                                   private val shopReviewHolderListener: ReviewShopViewHolder.ShopReviewHolderListener) : ReviewProductTypeFactoryAdapter(viewListener) {
    override fun createViewHolder(parent: View, type: Int): AbstractViewHolder<*> {
        return if (type == ReviewShopViewHolder.Companion.LAYOUT) {
            ReviewShopViewHolder(parent, viewListener, shopReviewHolderListener)
        } else {
            super.createViewHolder(parent, type)
        }
    }

    fun type(shopReviewModelContent: ReviewShopModelContent?): Int {
        return ReviewShopViewHolder.Companion.LAYOUT
    }

}